src/commandsAndMenu.tsx

```diff
@@ -407,11 +419,26 @@ export function addCommands(
           'Encountered an error when pulling changes. Error: ',
           error
         );
-        logger.log({
-          message: trans.__('Failed to pull'),
-          level: Level.ERROR,
-          error
-        });
+
+        const errorMsg = typeof error === 'string' ? error : error.message;
+
+        // Discard changes then retry pull
+        if (
+          errorMsg
+            .toLowerCase()
+            .includes(
+              'your local changes to the following files would be overwritten by merge'
+            )
+        ) {
+          await discardAllChanges(gitModel, trans, true);
+          await commands.execute(CommandIDs.gitPull);
```

Comment:
We need to rework this a bit to handle cancel action.

```suggestion
          await commands.execute(CommandIDs.gitPull, {discard: true, fallback: true});
```